pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/v2/store/StarTreeLoaderUtils.java

```diff
@@ -100,7 +100,16 @@ public StarTreeV2Metadata getMetadata() {
 
         @Override
         public DataSource getDataSource(String columnName) {
-          return dataSourceMap.get(columnName);
+          DataSource result = dataSourceMap.get(columnName);
+          // Some query columnNames could be supported by the underlying value aggregation type; do a secondary lookup
```

Comment:
During query time, can we first resolve it instead of relying on the fallback? Ideally we should resolve it before looking up the metadata

Reply:
Resolving before looking up the metadata is a little tricky due to possible knock-on effects to the StarTreeIndexReader. I haven't been able to formulate a straightforward solution (yet) - any ideas would be welcome.
Also, I reverted to checking the columnName directly against the map without resolving stored types because this is necessary for group by queries where the columnName might be a dimension, not necessarily a function column pair.

Reply:
I found a way to accomplish the above without having to make any changes to the StarTreeIndexReader.